doc/index.md

```diff
@@ -291,6 +291,101 @@ sentinel. That is only necessary if the underlying range's sentinel type models
 BidirectionalIterator. That's a finer point that you shouldn't worry about right
 now.)*
 
+## Create Custom Iterators
+
+Here is example of Range v3 compatible RandomAcess proxy iterator.
+Iterator return std::pair key/value (aka zip).
+
+~~~~~~~{.cpp}
+    using KeyIter   = typename std::vector<Key>::iterator;
+    using ValueIter = typename std::vector<Value>::iterator;
+
+    struct cursor {
+
+        // basic_iterator derives from "mixin", if present, so it can be used
+        // to inject things into the public interface of the iterator
+        struct mixin : ranges::basic_mixin<cursor>
+        {
+            using ranges::basic_mixin<cursor>::basic_mixin;
+
+            // It is necessary to expose constructor in this way
+            mixin(KeyIter key_iterator, ValueIter value_iterator)
+                : mixin{ cursor(key_iterator, value_iterator) }
+            {}
+
+            KeyIter key_iterator() {
+                return this->get().key_iterator;
+            }
+            ValueIter value_iterator() {
+                return this->get().value_iterator;
+            }
+        };
+
+        // This is for dereference operator.
+        using cursor_value_t = std::pair<Key, Value&>;
+        cursor_value_t read() const {
+            return {*key_iterator, *value_iterator};
+        }
+
+        // arrow for operator-> . It is not required.
+        struct arrow_proxy
+        {
+            cursor_value_t value;
+            cursor_value_t* operator->() { return std::addressof(value); }
+            operator cursor_value_t*()   { return std::addressof(value); }
+        };
+        arrow_proxy arrow() const {
+            return arrow_proxy{ read() };
+        }
```

Comment:
Remove this support for `operator->`. `it->x` is supposed to be semantically equivalent to `(*it).x`, and it is not in this case.

Reply:
Why `it->x` is not equivalent to `(*it).x` ?
`iterator->value` points to the same place as `(*iterator).value` ...